src/components/Hoverable/index.js

```diff
@@ -1,147 +1,101 @@
-                // and the wrapperView contains the element that caused the blur and reset isHovered
-                if (!this.wrapperView.contains(el.target) && !this.wrapperView.contains(el.relatedTarget)) {
-                    this.setIsHovered(false);
-                }
-
-                if (_.isFunction(child.props.onBlur)) {
-                    child.props.onBlur(el);
-                }
-            },
-        });
-    }
+
+function Hoverable({disabled, onHoverIn, onHoverOut, children}) {
+    const [isHovered, setIsHovered] = useState(false);
+    const wrapperView = useRef(null);
+
+    useEffect(() => {
+        // For now, listener is quite simple so we don't need to extract its declaration from effect
+        const onVisibilityChange = () => document.visibilityState === 'hidden' && setIsHovered(false);
+
+        document.addEventListener('visibilitychange', onVisibilityChange);
+
+        return () => document.removeEventListener('visibilitychange', onVisibilityChange);
+    }, []);
+
+    useEffect(() => {
+        if (disabled) return setIsHovered(false);
+    }, [disabled]);
+
+    useEffect(() => {
+        if (isHovered) return onHoverIn();
+        onHoverOut();
+    }, [isHovered, onHoverIn, onHoverOut]);
+
+    const child = useMemo(() => React.Children.only(mapChildren(children, isHovered)), [children, isHovered]);
+
+    const refCallback = useCallback(
+        (el) => {
+            if (!el) return;
+
+            wrapperView.current = el;
+
+            const {ref} = child;
+
+            if (_.isFunction(ref)) return ref(el);
+
+            if (_.isObject(ref)) return (ref.current = el);
+        },
+        [child],
+    );
+
+    const onMouseEnter = useCallback(
+        (el) => {
+            setIsHovered(true);
+
+            if (_.isFunction(child.props.onMouseEnter)) child.props.onMouseEnter(el);
+        },
+        [child],
+    );
+
+    const onMouseLeave = useCallback(
+        (el) => {
+            setIsHovered(false);
+
+            if (_.isFunction(child.props.onMouseLeave)) child.props.onMouseLeave(el);
+        },
+        [child],
+    );
+
+    const onBlur = useCallback(
+        (el) => {
+            // Check if the blur event occurred due to clicking outside the element
+            // and the wrapperView contains the element that caused the blur and reset isHovered
+            if (!wrapperView.current.contains(el.target) && !wrapperView.current.contains(el.relatedTarget)) {
+                setIsHovered(false);
+            }
+
+            if (_.isFunction(child.props.onBlur)) child.props.onBlur(el);
+        },
+        [child],
+    );
+
+    if (!DeviceCapabilities.hasHoverSupport()) return child;
+
+    return React.cloneElement(child, {
+        ref: refCallback,
+        onMouseEnter,
+        onMouseLeave,
+        onBlur,
+    });
 }
 
 Hoverable.propTypes = propTypes;
 Hoverable.defaultProps = defaultProps;
```

Comment:
Add a [displayName field ](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#function-component-style)
```
Hoverable.displayName = 'Hoverable';
```

Reply:
AFAIK, [there is no need to set `displayName` explicitly](https://legacy.reactjs.org/docs/react-component.html#displayname) on components. Should we propose an update to the style doc you've mentioned?

Reply:
Yeah, but this standard is from Expensify codebase

Reply:
Sure, let's keep it this way then :)